### PR TITLE
localstorage vault

### PR DESCRIPTION
**Pull request by pldespaigne**

### CODE REVIEW

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      throw new Error('This environement doesn\'t support a Local Storage');
+    }
+  }
+
+  get(accounts: 'accounts'): Promise<string[]>;
+  get(address: 'default'): Promise<string>;
+  // tslint:disable-next-line: unified-signatures
+  get(address: string): Promise<string>;
+  get(key: KeyVault): Promise<string | string[]> {
+    switch(key) {
```
Comment (GrandSchtroumpf) [inline libs/ngeth/src/lib/wallet/localstorage-vault.ts]: redundant code. Parse only for accounts (array).

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      default: return JSON.parse(this.storage.getItem(key));
+    }
+  }
+
+  set(key: 'accounts', accounts: string[]): Promise<any>;
+  set(key: 'default', address: string): Promise<any>;
+  // tslint:disable-next-line: unified-signatures
+  set(address: string, json: string): Promise<any>;
+  set(key: string, payload: string | string[]): Promise<any> {
+    return new Promise((res, rej) => {
```
Comment (GrandSchtroumpf) [inline libs/ngeth/src/lib/wallet/localstorage-vault.ts]: Just add "async" in front of the function, it'll transform the result into a promise (no need to do `new Promise`).

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      }
+      if (key !== 'accounts' && Array.isArray(payload)) {
+        rej('When key is "default" or an address, payload must be a string');
+      }
+
+      res(this.storage.setItem(key, JSON.stringify(payload)));
+    });
+  }
+
+  delete(key: KeyVault): Promise<any> {
```
Comment (GrandSchtroumpf) [inline libs/ngeth/src/lib/wallet/localstorage-vault.ts]: Save as above

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return new Promise(res => {
+      res(this.storage.removeItem(key));
+    })
+  }
+
+  clear(): Promise<any> {
```
Comment (GrandSchtroumpf) [inline libs/ngeth/src/lib/wallet/localstorage-vault.ts]: Save as above

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    return new Promise(res => {
+      this.get('accounts').then(accounts =>
```
Comment (GrandSchtroumpf) [inline libs/ngeth/src/lib/wallet/localstorage-vault.ts]: Localstorage as a method "clear"

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/ngeth/src/lib/wallet/localstorage-vault.ts (hunk 5 above)
Comment (pldespaigne) [reply libs/ngeth/src/lib/wallet/localstorage-vault.ts]: yes but we don't want to clear the whole localStorage of the user, but only the items belonging to the vault

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**